packages/filesystem/src/electron-browser/file-dialog/electron-file-dialog-service.ts

```diff
@@ -94,10 +94,17 @@ export class ElectronFileDialogService extends DefaultFileDialogService {
     protected toDialogOptions(uri: URI, props: SaveFileDialogProps | OpenFileDialogProps, dialogTitle: string): electron.FileDialogProps {
         const title = props.title || dialogTitle;
         const defaultPath = FileUri.fsPath(uri);
-        const filters: FileFilter[] = [{ name: 'All Files', extensions: ['*'] }];
+        const allFilesFilter = { name: 'All Files', extensions: ['*'] };
+        const filters: FileFilter[] = [];
+        if (props.allFilesPosition === undefined || props.allFilesPosition === FileDialogTreeAllFilesPosition.FIRST) {
+            filters.push(allFilesFilter);
+        }
         if (props.filters) {
             filters.push(...Object.keys(props.filters).map(key => ({ name: key, extensions: props.filters![key] })));
         }
+        if (props.allFilesPosition === FileDialogTreeAllFilesPosition.LAST) {
```

Comment:
You have a `this.fileDialogOptions !== undefined && ` check [in the browser-based service](https://github.com/theia-ide/theia/pull/4820/files#diff-60575443566d850b803281871cc077b0R63). Shouldn't we align the code?

Reply:
Not really, the equivalent would be checking that `props` is not `undefined`, which is already checked by typescript.

Reply:
In your browser code, if `this.fileDialogOptions` is defined but `this.fileDialogOptions.allFilesPosition` is `udnefined`, what should happen?

Reply:
`All Files` should be shown as the first option, since this is the current behaviour of Theia.

Reply:
Sorry, help me out with this please, I must be blind, where do you add the `allFilesElement` to the `fileTypes` array, if `this.fileDialogOptions` is defined but `this.fileDialogOptions.allFilesPosition` is `undefined`.

Reply:
Sorry I thought you meant a different thing. Yes, the case where `this.fileDialogOptions.allFilesPosition` is `undefined` is not correctly managed. I'll fix that. Thanks for pointing that out.